src/Microsoft.AspNet.Identity/TaskExtensions.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNet.Identity
+{
+    public static class TaskExtensions
+    {
+        public static async Task<IdentityResult> WithLoggingAsync<TUser>(this Task<IdentityResult> identityResult, UserManager<TUser> userManager,TUser user,
+            [System.Runtime.CompilerServices.CallerMemberName] string methodName = "") where TUser :class
+        {
+            return await (await identityResult).WithLoggingAsync(userManager, user, methodName);
```

Comment:
Yeah I said the same thing about the "this" smell. Also result extensions is no longer extensible unlike the virtual method. If they add their own derived sign in result with a new status code there is no way to update the logging extension method.